### ISSUE TITLE
Hard coded DigitCaps num_outputs?

### ISSUE THREAD

**Issue by Allong12** [naturomics/CapsLayer#10]
https://github.com/naturomics/CapsLayer/blob/1709caceccd183bf800262ade12729278cc12165/models/vectorCapsNet.py#L68

Currently, the `num_outputs` DigitCaps fully connected layer is hard coded to 10, which I believe is to represent the number of logits found in MNIST, in consideration of model portability, should this perhaps be set to `num_outputs=self.num_label`?

While on the topic, I don't believe I understand the 16 in `out_caps_shape=[16, 1]`, but I haven't really grokked the original paper either, is there some performance or accuracy benefit in twiddling with the out_caps_shape parameter?

**Comment by naturomics** [comment]
You're right, `num_outputs=self.num_label`. I'm rebuilding a lot of code recently, this is changed as you say. Hope I have time to finish a new version and release it.

About the `out_caps_shape`, it's for compatibility. Consider this situation, now the digit capsules output a 16x1 vector, can we change them to output 2x8, even a tensor [2, 2, 2]? Of course we can, this `out_caps_shape` parameter will give us a great convenience to do test on different capsule output shape
